tools/wptrunner/wptrunner/executors/protocol.py

```diff
@@ -605,3 +606,56 @@ def load_reftest_analyzer(self, test, result):
         self.parent.base.load(urljoin(self.parent.executor.server_url("https"),
                               "/common/third_party/reftest-analyzer.xhtml#log=%s" %
                                quote(output.getvalue())))
+
+
+class ConnectionlessBaseProtocolPart(BaseProtocolPart):
+    def load(self, url):
+        pass
+
+    def execute_script(self, script, asynchronous=False):
+        pass
+
+    def set_timeout(self, timeout):
+        pass
+
+    def wait(self):
+        return False
+
+    def set_window(self, handle):
+        pass
+
+    def window_handles(self):
+        return []
+
+
+class ConnectionlessProtocol(Protocol):
+    implements = [ConnectionlessBaseProtocolPart]
+
+    def connect(self):
+        pass
+
+    def after_connect(self):
+        pass
+
+
+class WdspecProtocol(ConnectionlessProtocol):
```

Comment:
Isn't it weird to have `WdspecProtocol` inherit from `ConnectionlessProtocol`? It used to be `Protocol`.

Sure, subclasses can add things, but putting "connectionless" in the name does suggest that not having a connection is a core aspect of this and its subclasses.

Would it work to just implement `connect` and `after_connect` in `WdspecProtocol` instead?

Reply:
I see, I think the `ConnectionlessProtocol` name has confused me. Is the "connection" a direct connection between wptrunner and the browser, and not just any connection? I found this strange because WebDriver obviously involves a HTTP connection.